### PR TITLE
Improve-enrollment-errors

### PR DESCRIPTION
**Pull request by bjkeller**
Adds validation error handling to the creation of enrollment and transfer records.

Also adds code that allows dates to be parsed with more than one format.

### CODE REVIEW

File: common/src/python/dates/form_dates.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+"""Utilities to handle dates."""
+
+from datetime import datetime
+from typing import List
+
+DATE_FORMATS = ['%m/%d/%Y', '%Y-%m-%d']
```
Comment (hewanadun) [inline common/src/python/dates/form_dates.py:6]: QC rules accept either - or /, so all 4 formats '%m/%d/%Y', '%m-%d-%Y', '%Y/%m/%d', '%Y-%m-%d' are accepted

File: common/test/python/enrollment/test_serialization.py [hunk 2]
```diff
@@ -1,9 +1,23 @@
...
 from pydantic import ValidationError
 
 
+@pytest.fixture
+def bad_date_row():
+    yield {
+        'adcid': 0,
+        'ptid': "123456",
+        'naccid': "000000",
+        'frmdate_enrl': "10/06/2024",
```
Comment (hewanadun) [inline common/test/python/enrollment/test_serialization.py:16]: Isn't this a valid date format?

**Review by hewanadun — DISMISSED** [review]
Please see the comments below.

File: common/src/python/dates/form_dates.py (hunk 1 above)
Comment (bjkeller) [reply common/src/python/dates/form_dates.py:6]: changed

File: common/test/python/enrollment/test_serialization.py (hunk 2 above)
Comment (bjkeller) [reply common/test/python/enrollment/test_serialization.py:16]: yes. removed the slashes

**Pull request merged**